modules/social_features/social_group/src/Plugin/Block/GroupAddBlock.php

```diff
@@ -28,7 +28,7 @@ public function blockAccess(AccountInterface $account) {
     $route_user_id = \Drupal::routeMatch()->getParameter('user');
 
     // Show this block only on current user Groups page.
-    if ($current_user->id() == $route_user_id) {
+    if ($current_user->id() == $route_user_id && ($account->hasPermission("create open_group group") || $account->hasPermission("creat closed group") || $account->hasPermission("create public_group group"))) {
```

Comment:
Do you think it will also be possible to make this access check a bit more generic? E.g. what if websites have disabled all the group types and created own custom group type? Is there a way to retrieve all the group types and then check if the user has permission to create new groups of one of the group types?

Reply:
I'll look into it

Reply:
@slowflyer  GroupType::loadMultiple give you all group types available... then you could check it for all available groups very easily.
    foreach (GroupType::loadMultiple() as $group_type) {
      $permissions = 'create ' . $group_type->label() . 'group'; //or id();
}